src/active-response/firewalls/default-firewall-drop.c

```diff
@@ -91,15 +91,15 @@ int main (int argc, char **argv) {
         if (access(iptables_tmp, F_OK) < 0) {
             char iptables_path[COMMANDSIZE_4096];
             memset(iptables_path, '\0', COMMANDSIZE_4096);
-            snprintf(iptables_path, COMMANDSIZE_4096 - 1, "/usr%s", iptables_tmp);
+            snprintf(iptables_path, COMMANDSIZE_4096, "/usr%s", iptables_tmp);
             if (access(iptables_path, F_OK) < 0) {
                 memset(log_msg, '\0', OS_MAXSTR);
                 snprintf(log_msg, OS_MAXSTR -1, "The iptables file '%s' is not accessible: %s (%d)", iptables_path, strerror(errno), errno);
                 write_debug_file(argv[0], log_msg);
                 cJSON_Delete(input_json);
                 return OS_SUCCESS;
             }
-            strncpy(iptables, iptables_path, COMMANDSIZE_4096 - 1);
+            strncpy(iptables, iptables_path, COMMANDSIZE_4096);
```

Comment:
Hi @RamiroRD 

The reason for this warning is that the compiler tells us that we are always truncating the string without copying the null-terminated string, it is a particular case because the source and destination buffer sizes are the same.

Best regards
Hanes